roles/openshift_hosted/tasks/registry/registry.yml

```diff
@@ -64,13 +64,19 @@
   when:
   - openshift.hosted.registry.storage.kind | default(none) == 'object'
 
-- name: Set facts for persistent volume
-  set_fact:
-    pvc_volume_mounts:
-    - name: registry-storage
-      type: persistentVolumeClaim
-      claim_name: "{{ openshift.hosted.registry.storage.volume.name }}-claim"
-    openshift_hosted_registry_volumes: "{{ openshift_hosted_registry_volumes | union(pvc_volume_mounts) }}"
+- block:
+
+  - name: Set facts for persistent volume
+    set_fact:
+      pvc_volume_mounts:
+      - name: registry-storage
+        type: persistentVolumeClaim
+        claim_name: "{{ openshift.hosted.registry.storage.volume.name }}-claim"
+
+  - name: Update openshift_hosted facts for persistent volumes
+    set_fact:
+      openshift_hosted_registry_volumes: "{{ openshift_hosted_registry_volumes | union(pvc_volume_mounts) }}"
```

Comment:
I think we should consider refactoring this down the line to avoid needing to use set_fact in the way that we are, but I do think this could be reduced down to a single set_fact instead of two separate set_facts:

```
- name: Update openshift_hosted facts for persistent volumes
  set_fact:
    openshift_hosted_registry_volumes: "{{ openshift_hosted_registry_volumes | union(pvc_volume_mounts) }}"
  vars:
    pvc_volume_mounts:
    - name: registry-storage
      type: persistentVolumeClaim
      claim_name: "{{ openshift.hosted.registry.storage.volume.name }}-claim"
```